lib/jsdom/living/nodes/HTMLCanvasElement-impl.js

```diff
@@ -50,16 +82,47 @@ class HTMLCanvasElementImpl extends HTMLElementImpl {
   }
 
   toDataURL() {
-    if (this._nodeCanvas) {
-      return this._nodeCanvas.toDataURL.apply(this._nodeCanvas, arguments);
+    const canvas = this._getCanvas();
+    if (canvas) {
+      return canvas.toDataURL.apply(this._canvas, arguments);
     }
 
     notImplemented("HTMLCanvasElement.prototype.toDataURL (without installing the canvas npm package)",
       this._ownerDocument._defaultView);
   }
 
+  toBlob(callback, type, encoderOptions) {
+    const window = this._ownerDocument._defaultView;
+    const canvas = this._getCanvas();
+    if (canvas) {
+      let stream;
+      switch (type) {
+        case "image/jpg":
+        case "image/jpeg":
+          stream = canvas.createJPEGStream({
+            quality: Math.min(0, Math.max(1, encoderOptions)) * 100
+          });
+          break;
+        default:
+          // TODO: Patch node-canvas to receive encoderOptions for PNG stream
+          type = "image/png";
+          stream = canvas.createPNGStream();
+      }
+      const buffers = [];
+      stream.on("data", chunk => {
+        buffers.push(chunk);
+      });
+      stream.on("end", () => {
+        callback(new window.Blob(buffers, { type }));
+      });
+    } else {
+      notImplemented("HTMLCanvasElement.prototype.toBlob (without installing the canvas npm package)",
+        window);
+    }
+  }
+
   get width() {
-    const parsed = parseInt(this.getAttribute("width"));
+    const parsed = parseInt(this.getAttribute("width"), 10);
```

Comment:
ES5 removed octal interpretation. older browsers will indeed interpret it as octal, however those older browsers are not supported anyway, since we require ES6 features. Do note that: `parseInt('0x123', 10) === 0` and `parseInt('0x123') === 291`